src/Runner.js

```diff
@@ -168,27 +179,28 @@ export default class Runner extends EventEmitter {
     }
     if (process.platform === 'win32') {
       // Windows doesn't exit the process when it should.
-      spawn('taskkill', ['/pid', '' + this.debugprocess.pid, '/T', '/F']);
+      spawn('taskkill', ['/pid', `${this.debugprocess.pid}`, '/T', '/F']);
     } else {
       this.debugprocess.kill();
     }
     delete this.debugprocess;
   }
 
+  // eslint-disable-next-line class-methods-use-this
   findMessageType(buf: Buffer): MessageType {
-    const str = buf.toString('utf8', 0, 58);
-    const lastCommitRegex = /No tests found related to files changed since ((last commit)|("[a-z0-9]+"))./;
-    if (lastCommitRegex.test(str)) {
-      return messageTypes.noTests;
-    }
-    if (/^\s*Watch Usage\b/.test(str)) {
-      return messageTypes.watchUsage;
-    }
-
-    if (str.trim().startsWith('Test results written to')) {
-      return messageTypes.testResults;
-    }
-    return messageTypes.unknown;
+    const noTestRegex = /No tests found related to files changed since ((last commit)|("[a-z0-9]+"))./;
+    const watchUsageRegex = /^\s*Watch Usage\b/;
+    const testResultsRegex = /Test results written to/;
+
+    const checks = [
+      [testResultsRegex, messageTypes.testResults],
+      [noTestRegex, messageTypes.noTests],
+      [watchUsageRegex, messageTypes.watchUsage],
+    ];
+
+    const str = buf.toString('utf8');
+    const check = checks.find(_check => _check[0].test(str));
+    return check ? check[1] : messageTypes.unknown;
```

Comment:
The leading underscore and `[0]` in this line caught my eye:
```js
    const check = checks.find(_check => _check[0].test(str));
```

How do you feel about naming these?
```js
    const checks = [
      { regex: testResultsRegex, messageType: messageTypes.testResults },
      { regex: noTestRegex, messageType: messageTypes.noTests },
      { regex: watchUsageRegex, messageType: messageTypes.watchUsage },
    ];

    const str = buf.toString('utf8');
    const match = checks.find(({ regex })=> regex.test(str));
    return match ? match.messageType : messageTypes.unknown;
```

Reply:
good, done.